cns/restserver/util.go

```diff
@@ -40,6 +46,49 @@ func (service *HTTPRestService) setNetworkInfo(networkName string, networkInfo *
 	service.state.Networks[networkName] = networkInfo
 }
 
+func (service *HTTPRestService) SetPnpIDMacaddressMapping() error {
+	service.Lock()
+	defer service.Unlock()
+	VfMacAddressMapping, err := fetchMacAddressPnpIDMapping()
+	if err != nil {
+		return err
+	}
+	service.PnpIDByMacAddress = VfMacAddressMapping
+	return nil
+}
+
+func (service *HTTPRestService) getPNPIDFromMacAddress(macAddress string) (string, error) {
+	if _, ok := service.PnpIDByMacAddress[macAddress]; !ok {
+		if err := service.SetPnpIDMacaddressMapping(); err != nil {
+			return "", err
+		}
+	}
+	return service.PnpIDByMacAddress[macAddress], nil
+}
+
+func fetchMacAddressPnpIDMapping() (map[string]string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), Timeout)
+	defer cancel() // The cancel should be deferred so resources are cleaned up
+	cmd := exec.CommandContext(ctx, "powershell", GetMacAddressVFPPnpIDMapping)
+	output, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[string]string)
+	if len(output) == 0 {
+		return nil, errors.New("Network adapter not found")
```

Comment:
Why do we error out here? Can't we have an empty mapping if IB is not available in the node?

Reply:
Addressed